### ISSUE TITLE
Should we use Yarn instead of npm?

### ISSUE THREAD

**Issue by wholesomedev** [hollowverse-archive/hollowverse#56]
@levsthings suggested that we start a discussion with regard whether we should use Yarn instead of npm.

Yarn seems to be better than npm. But there are pros and cons to each package manager:

### Pros

* Faster installs
* Automatic dependency locking
* Nice and sensible CLI commands

### Cons

* Not as many people are comfortable with it as with npm
* Will probably need to exist alongside npm as we won't be able to completely get rid of npm (this is an assumption I'm making)
* New, uncertain future

### Pros

* Familiar to most people
* Established, and the whole _npm, Inc_ has a vested interest in its success
* Can be configured to lock dependencies

### Cons

* Slow installs
* Annoying CLI commands with `--save-dev`, `--save` etc
* No automatic dependency locking

---

Again, Yarn seems better than npm at this moment. However, my personal preference is to stick to npm for the following reasons (order of importance):

1. I don't want to add another tool to our toolset for marginal gains. This would especially be an issue if using Yarn means we still have to keep npm around for some tasks...
1. I don't have as much confidence in Yarn as I do in npm
1. npm has learned from Yarn and is improving, see these tweets:
    * https://twitter.com/notquiteleo/status/855562823199600641
    * https://twitter.com/maybekatz/status/855362606713851904
    * https://twitter.com/maybekatz/status/849380703469002752

Anyway, with all that said, I'm still willing to give Yarn a try.

**Comment by levsthings** [comment]
I agree with most of the points but I'd like to expand on the pros section of npm.

> npm
> Pros
> 
>     Familiar to most people
>     Established, and the whole npm, Inc has a vested interest in its success
>     Can be configured to lock dependencies
> 

- While npm is familiar to most of the people, a lot of projects which exist on the latest React paradigm and TypeScript almost always use Yarn. So there's a good chance anyone who's interested in our project will be comfortable with Yarn. I wouldn't mention this if we had, say, an Angular project.
- There's no doubt about npm's vested interest and I see that they are going to catch up with Yarn eventually. But we should keep in mind that npm is serving to the entire JavaScript community where as Yarn is mostly used by projects like ours. It's safe to say that it serves our niche.
- As we discussed, we can still lock dependencies with `shrinkwrap` but there's information out there that seems to suggest that yarn's deterministic locking feature is better than `shrinkwrap`. **See** http://stackoverflow.com/questions/40057469/what-is-the-difference-between-yarn-lock-and-npm-shrinkwrap

Having said these, I should also tell that I resisted using Yarn in my projects for a very long time, I really didn't like the community separation aspect and I yielded only after realizing it really is just a better tool.

I say let's go for it. After `npm 5` is released and tested out in the wild, we can always go back. For now though, we could really use faster CI builds and a tested `.lock` mechanism.
